utils/zabbix/utils.go

```diff
@@ -0,0 +1,115 @@
+package zabbix
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+)
+
+type ZabbixAPI struct {
+	URL      string
+	Key      string
+	User     string
+	Password string
+}
+
+// Initialize gets Zabbix username, password and URL
+// from the environment and sets them in ZabbixAPI struct
+func (z *ZabbixAPI) Initialize() {
+	user := os.Getenv("ZABBIX_USER")
+	if user == "" {
+		log.Fatalf("ENV ZABBIX_USER not set in environment.")
+	} else {
+		z.User = user
+	}
+
+	password := os.Getenv("ZABBIX_PASSWORD")
+	if password == "" {
+		log.Fatalf("ENV ZABBIX_PASSWORD not set in environment.")
+	} else {
+		z.Password = password
+	}
+
+	url := os.Getenv("ZABBIX_URL")
+	if url == "" {
+		log.Fatalf("ENV ZABBIX_URL not set in environment.")
+	} else {
+		z.URL = url
+	}
+}
+
+// MakeAPIRequest makes an API call to Zabbix.
+// It returns a response object and an error object.
+// ZABBIX_URL needs to be set in the env for this to work.
+func (z *ZabbixAPI) MakeAPIRequest(payload map[string]interface{}) (resp *http.Response, err error) {
+	// Build the request
+	b, err := json.Marshal(payload)
+	if err != nil {
+		log.Fatalf("Cannot marshal payload. Error: %v\n", err)
+	}
+
+	body := bytes.NewReader(b)
+	req, err := http.NewRequest("GET", z.URL, body)
+	if err != nil {
+		log.Fatalf("Cannot make HTTP request. Error: %v\n", err)
+	}
+
+	req.Header.Set("Content-type", "application/json")
+	// Send the request via a client
+	resp, err = http.DefaultClient.Do(req)
+	return resp, err
+}
+
+func (z *ZabbixAPI) getAPIKey() (string, error) {
```

Comment:
What do you think about checking the struct field for the API Key and returning it if it is set?

Reply:
good idea. will do that.